src/components/plots/FeaturesVsUMIsScatterplot.jsx

```diff
@@ -3,23 +3,29 @@ import PropTypes from 'prop-types';
 import { Vega } from 'react-vega';
 
 import EmptyPlot from './helpers/EmptyPlot';
+import transformOldFeaturesVsUMIsPlotData from './helpers/transformOldFeaturesVsUMIsPlotData';
 import generateSpec from '../../utils/plotSpecs/generateFeaturesVsUMIsScatterplot';
 
 const FeaturesVsUMIsScatterplot = (props) => {
   const {
     config, plotData, actions,
   } = props;
 
+  // can remove if we formally migrate
+  const newPlotData = (!Array.isArray(plotData) || !plotData.length)
+    ? plotData
+    : transformOldFeaturesVsUMIsPlotData(plotData);
```

Comment:
This is very difficult to read and understand. Can you rewrite these 3 lines to be easier to read? I suggest you break them down into multiple statements.

Reply:
I tried to make this more clear but let me know if it you had something else in mind

Reply:
it looks better :)